xdsl/ir/core.py

```diff
@@ -870,10 +878,24 @@ def clone(
         return op
 
     @classmethod
-    def has_trait(cls, trait: type[OpTrait], parameters: Any = None) -> bool:
+    def has_trait(
```

Comment:
not sure if we want a `maybe_has_trait` with this functionality and leave `has_trait` as it was.

Reply:
Yes, I think that your approach makes sense!